File: server/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/UserEnabledUpdateRequest.java
package edu.cnm.deepdive.seesomethingabq.model.dto;

/**
 * Request body DTO for updating enabled/disabled status of a user.
 */
public class UserEnabledUpdateRequest {

  private boolean enabled;

  /**
   * Returns the desired enabled/disabled state for the user.
   *
   * @return {@code true} to enable; {@code false} to disable.
   */
  public boolean isEnabled() {
    return enabled;
  }

  /**
   * Sets the desired enabled/disabled state for the user.
   *
   * @param enabled {@code true} to enable; {@code false} to disable.
   */
  public void setEnabled(boolean enabled) {
    this.enabled = enabled;
  }

}

File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/UserEnabledUpdateRequest.kt
package edu.cnm.deepdive.seesomethingabq.model.dto

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Request body for PATCH /manager/users/{externalId}/enabled.
 *
 * Server expects a JSON object with a single boolean property named "enabled".
 */
data class UserEnabledUpdateRequest(
  @Expose
  @SerializedName("enabled")
  val enabled: Boolean,
)

